lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -357,6 +345,90 @@ def from_string(private_key: str) -> "PrivateKey":
         return PrivateKey(raw=private_key.strip())
 
 
+@dataclass(frozen=True)
+class Certificate:
+    """This class represents a certificate."""
+
+    raw: str
+    common_name: str
+    sans_dns: Optional[List[str]] = None
+    sans_ip: Optional[List[str]] = None
+    sans_oid: Optional[List[str]] = None
```

Comment:
Because the dataclass is frozen, it would make sense to make these tuples instead to keep them immutable. Otherwise, it will still be possible to mutate those lists after the object creation.

Reply:
This applies elsewhere in code that is already merged also.

Reply:
done